test/e2e/e2e_test.go

```diff
@@ -0,0 +1,107 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/engine"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+)
+
+// These tests confirm that all container and operator policies properly pass
+// with a known-good image, and properly fail with a known-bad image.
+// Any check that is found in the error section of the Result will cause this
+// to fail.
+var _ = Describe("policy validation", func() {
+	Describe("When enforcing operator policy", func() {
+		var (
+			// TODO: replace this with a project-accessible namespace once identified.
+			goodImage = "quay.io/komish/preflight-test-bundle-passes:latest"
+			badImage  = "quay.io/komish/preflight-test-bundle-fails:latest"
+		)
+
+		Context("with a known-good image", func() {
+
+			cfg := runtime.Config{
+				Image:         goodImage,
+				EnabledChecks: engine.OperatorPolicy(),
+			}
+
+			engine, err := engine.NewForConfig(cfg)
+			Expect(err).To(BeNil())
+
+			engine.ExecuteChecks()
+			results := engine.Results()
+
+			It("should pass all checks", func() {
+				Expect(len(results.Passed)).To(Equal(len(cfg.EnabledChecks)))
+			})
+		})
+
+		Context("with a known-bad image", func() {
+			cfg := runtime.Config{
+				Image:         badImage,
+				EnabledChecks: engine.OperatorPolicy(),
+			}
+
+			engine, err := engine.NewForConfig(cfg)
+			Expect(err).To(BeNil())
+
+			engine.ExecuteChecks()
+			results := engine.Results()
+
+			// TODO: Replace this check so that you test for individual check failures
+			It("should not pass any checks", func() {
+				Expect(len(results.Passed)).To(BeZero())
+			})
+		})
+	})
+
+	Describe("When enforcing container policy", func() {
+
+		var (
+			// TODO: replace this with a project-accessible namespace once identified.
+			goodImage = "quay.io/komish/preflight-test-container-passes:latest"
+			// badImage  = "quay.io/komish/preflight-test-container-fails:latest"
```

Comment:
If you do the below about XContext, then you would not have to comment this due to non-usage.

Reply:
> I would also like to see some Makefile entries to both building and pushing the test images.

I would have built this here, but we dont' have a quay namespace built for these assets, and currently the assets reside in my namespace. I've opened up a subsequent issue for this here: https://github.com/redhat-openshift-ecosystem/openshift-preflight/issues/59

Reply:
Ok. Will add more in the issue.